nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -831,6 +831,7 @@ void Costmap2DROS::getCostsCallback(
   unsigned int mx, my;
 
   Costmap2D * costmap = layered_costmap_->getCostmap();
+  std::unique_lock<Costmap2D::mutex_t> lock(*(costmap->getMutex()));
```

Comment:
One option would also be to lock and unlock closer to where the costmap is queried to reduce the time it's locked but I don't know if that time difference is really significant

Reply:
i think this is fine. If we wanted to optimize for that here, what we could do is restructure the loop to do all the TF calls first in batch, then lock and loop over the footprint cost getting. TF is the big operation that would be wasted with respect to the costmap locking

Reply:
Okay, for lack of time right now, will skip that